.github/workflows/regression.yml

```diff
@@ -1,84 +1,228 @@
+#    - on demand in a PR, by either of the following comments:
+#      - '/ci-run-regression'
+#      - '/ci-run-all'
+#      (the comment issuer must be a member of the Vector GH team)
 #
 # This workflow runs our regression detection experiments, which are relative
-# evaluations of the base SHA for the PR to whatever SHA was just pushed into
-# the project (unless that SHA happens to be master branch HEAD). The goal is to
-# give quick-ish feedback on all-up Vector for a variety of configs as to
-# whether throughput performance has gone down, gotten more variable in the
+# evaluations of the base SHA and head SHA, whose determination depends on how
+# the workflow is invoked.
+#
+# The goal is to give quick-ish feedback on all-up Vector for a variety of configs
+# as to whether throughput performance has gone down, gotten more variable in the
 # pushed SHA.
 #
 # Regression detection is always done relative to the pushed SHA, meaning any
 # changes you introduce to the experiment will be picked up both for the base
-# SHA variant and your current SHA. Tags are SHA-SHA. The first SHA is the one
-# that triggered this workflow, the second is the one of the Vector being
-# tested. For comparison images the two SHAs are identical.
+# SHA variant and your current SHA.
+#
+# Docker image tags are SHA-SHA. The first SHA is the one that triggered this
+# workflow, the second is the one of the Vector being tested.
+# For comparison images the two SHAs are identical.
 
-name: Regression Detector
+name: Regression Detection Suite
 
 on:
+  issue_comment:
+    types: [created]
+  merge_group:
+    types: [checks_requested]
+  # Don't want to run this on each PR commit, but because GH doesn't allow specifying different required checks
+  # for pull request and merge queue, we need to "run" it in pull request, but in the jobs we will just auto pass.
   pull_request:
-    paths-ignore:
-      - "docs/**"
-      - "rfcs/**"
-      - "website/**"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.issue.id || github.event.merge_group.base_sha || github.sha }}
+  cancel-in-progress: true
 
 jobs:
-  cancel-previous:
-    runs-on: ubuntu-22.04
-    timeout-minutes: 3
+
+  # Only run this workflow if files changed in areas that could possibly introduce a regression, or by valid comment
+  should-run:
+    runs-on: ubuntu-latest
+    if: github.event_name != 'issue_comment' || (github.event.issue.pull_request
+        && (contains(github.event.comment.body, '/ci-run-regression') || contains(github.event.comment.body, '/ci-run-all')))
+    outputs:
+      source_changed: ${{ steps.filter.outputs.source }}
+      comment_valid: ${{ steps.comment.outputs.isTeamMember }}
     steps:
-      - uses: styfle/cancel-workflow-action@0.11.0
-        with:
-          access_token: ${{ secrets.GITHUB_TOKEN }}
-          all_but_latest: true # can cancel workflows scheduled later
+    - uses: actions/checkout@v3
+
+    - name: Determine if source code changed
```

Comment:
If the check is skipped due to change detection, does it still get merged? Do we still want change detection now that is moved to the merge queue?

Reply:
> If the check is skipped due to change detection, does it still get merged?

In it's current form, if the trigger type is the merge queue, and changes to "source" are not detected, it will pass this check without running the regression experiments.

Originally the change detection we had was on push to master, it would not run if the changes were docs/website/rfcs .  I was here attempting to preserve some semblance of that, since there is no workflow_on::merge_group to filter paths.
My thought was that there wasn't really an opportunity to introduce a performance regression if the files changed were outside these paths.
But, if we want to be extra careful, I can def remove this job step to look for file changes.

Reply:
Gotcha, nah, that's fine. There seems to be a small risk that changes outside of these paths would affect performance. If we find that not to be the case in practice we can remove it.

I'm not sure if this is possible but it might be nice to invert this change detection logic. That is we provide paths where if the PR only touches those paths and no others it runs. This would reduce the likelihood that we introduce a new file path where we'd want to run these and forget to add it here.

Reply:
I'll take a look again. I wanted to do that when I made the changes, but recall not seeing a great way. But I'll have another look.

Reply:
I added some logic in a follow up step to handle reversing the logic to be more conservative about running it.